run_sunspider_like.c

```diff
@@ -0,0 +1,122 @@
+#include "quickjs/quickjs.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+double total = 0;
+
+int Execute(JSContext *ctx, const char *filename) {
+    FILE *file = fopen(filename, "r");
+
+    if (file == NULL) {
+        return 1;
+    }
+
+    fseek(file, 0, SEEK_END);
+    long file_size = ftell(file);
+    fseek(file, 0, SEEK_SET);
+
+    char *buffer = (char *)malloc(file_size + 1);
+    if (buffer == NULL) {
+        fclose(file);
+        fprintf(stderr, "Memory allocation error\n");
+        return 1;
+    }
+
+    size_t bytesRead = fread(buffer, 1, file_size, file);
+    fclose(file);
+
+    if (bytesRead != file_size) {
+        fprintf(stderr, "Error reading file: %s\n", filename);
+        printf("bytesRead: %zu, file_size: %ld\n", bytesRead, file_size);
+        free(buffer);
+        return 1;
+    }
+
+    buffer[file_size] = '\0';
+
+    clock_t start_time = clock();
+    JSValue ret_val = JS_Eval(ctx, buffer, file_size, filename, JS_EVAL_TYPE_GLOBAL);
+    clock_t end_time = clock();
+    double execution_time = ((double)(end_time - start_time)) / CLOCKS_PER_SEC;
+    total += execution_time;
+
+    if (JS_IsException(ret_val)) {
+        printf("Error %s\n", JS_ToCString(ctx, JS_GetException(ctx)));
+        JS_FreeValue(ctx, ret_val);
+        free(buffer);
+        return 1;
+    }
+
+    printf("File: %s, Execution Time: %.4f seconds\n", filename, execution_time);
+
+    JS_FreeValue(ctx, ret_val);
+    free(buffer);
+
+    return 0;
+}
+
+void RunTest(JSContext* ctx, char* path, char* buffer) {
+        char *fullpath = malloc(strlen(path) + strlen(buffer) + strlen("-data.js") + 1);
+
+        strcpy(fullpath, path);
+        strcat(fullpath, buffer);
+
+        strcat(fullpath, "-data.js");
+        if (Execute(ctx, fullpath) != 0) {
+            // Pass, only used for kraken.
+        }
+
+        fullpath[strlen(fullpath) - 8] = '\0';
+        strcat(fullpath, ".js");
```

Comment:
Just recompute it - more legible and anything's better than strcat :-)
```suggestion
    snprintf(fullpath, sizeof(fullpath), "%s%s.js", path, buffer);
```